pyfolio/tears.py

```diff
@@ -1567,10 +1576,13 @@ def create_perf_attrib_tear_sheet(returns,
             current_section += 1
 
         for factor_type, partitions in factor_partitions.iteritems():
+            columns_to_select = portfolio_exposures.columns.intersection(partitions)
+
+            if len(columns_to_select) == 0:
```

Comment:
This is still going to leave an empty subplot, right?  What's the difference between continuing here and passing an empty dataframe to `plot_risk_exposures`?

Reply:
Ah, I see, we're removing vertical sections above.

Reply:
I changed the call GridSpec, to match this the number of plots we call, so that there won't be an empty plot:
https://github.com/quantopian/pyfolio/pull/487/files#diff-135fcf845bbceefcd00ccb65ab24afc8R1547